Fix NPE in MetricsApiOtelCollectorTest.getApiMetrics()

riyaaroraa opened the pull request:
fixes #31971

- Add null check for client field before calling dumpMetrics()
- Initialize client if null to prevent NullPointerException
- Fixes GitHub issue #31971

riyaaroraa committed:
Fix NPE in MetricsApiOtelCollectorTest.getApiMetrics()

- Add null check for client field before calling dumpMetrics()
- Initialize client if null to prevent NullPointerException
- Fixes GitHub issue #31971

riyaaroraa commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=afe4dabd-6d3d-4066-9918-c6ac530cf946) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

fmhwong reviewed (changes requested):
From the issue #31971, the NPE was coming from `MetricsApiOtelCollectorTest.java` line 151.

```
List<String> splits = Arrays.asList(result.split("((?=# HELP))"));
```

If `result` was null, that would cause a NPE.

riyaaroraa committed:
Fix NPE - add null check for result from dumpMetrics()

- Add null check for result before calling result.split()
- Prevents NPE when client.dumpMetrics() returns null
- Addresses reviewer feedback on root cause of NPE at line 151

riyaaroraa commented:
!build
 ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=3ab80022-4dc9-44b1-a0cb-a66f800e0ed1) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

fmhwong reviewed (approved):
LGTM

The code change is good.  This fixed the NPE.  However, when no metrics were returned, this test case will fail.  Please investigate under what conditions that the `client.dumpMetrics()` will return `null` as a next step.  This is an intermittent failure so it won't be 100% reproducible in SOEs.
